### ISSUE TITLE
How to decode a Dictionary to get all locales of entry

### ISSUE THREAD

**Issue by mrigo-hfarm** [contentful/contentful.swift#202]
> - contentful.swift version number: 2.0.0
> - Xcode version number: 9.3.1
> - Target operating system(s) and version number(s)
    - [x] iOS: 
    - [ ] tvOS: 
    - [ ] watchOS:
    - [ ] macOS:
> - Package manager:
    - [ ] Carthage
    - [x] Cocoapods

Hi!
I want all locales for my entries, but I can't decode the result!

How can I decode a Dictionary as the resulting type returning from the api?

This is my function

```
let query = QueryOn<Contract>.where(field: Contract.Fields.type, .equals(type.rawValue)).localizeResults(withLocaleCode: "*")
            client?.fetchMappedEntries(matching: query) { (result: Result<MappedArrayResponse<Contract>>) in
                guard let contracts = result.value?.items else { return }
                if contracts.count > 0 {
                    contractsArr.append(contracts.first!)
                }
            }
```

and this is the class Contract

```
final class Contract: EntryDecodable, EntryQueryable {
    
    enum ContractType: String {
        case checkOut
        case checkIn
    }
    
    static let contentTypeId: String = "contract"
    
    let sys: Sys
    let type: String?
    let subject: [String: String]?
    let slugId: String?
    
    // Relationship fields.
    var categories: [String]?
    
    public required init(from decoder: Decoder) throws {
        
        sys              = try decoder.sys()
        let fields       = try decoder.contentfulFieldsContainer(keyedBy: Contract.Fields.self)
        
        self.type        = try fields.decodeIfPresent(String.self, forKey: .type)
        self.subject     = try fields.decodeIfPresent(Dictionary<String:String>.self, forKey: .subject)
        self.slugId      = try fields.decodeIfPresent(String.self, forKey: .slugId)
        self.categories  = try fields.decodeIfPresent(Array<String>.self, forKey: .categories)
        
    }
    
    enum Fields: String, CodingKey {
        case categories
        case type, subject, slugId
    }
}
```

The problem is, when I try to map the class Contrat, the field **subject** is returned like a Dictionary (and this it's ok!) but I can't decode as type [String: String].

How can I decode a Dictionary with the **MappedArrayResponse** type?

Thanks a lot!
\m

**Comment by loudmouth** [comment]
Hey @mrigo-hfarm 

I could add support for decoding responses for wildcard types in v3 of the SDK and there will be breaking changes for this next version.

**Comment by jroehl** [comment]
Closed due to inactivity and partially addressed in #225
